vision/product-search/cloud-client/src/test/java/com/example/vision/ProductSearchIT.java

```diff
@@ -0,0 +1,118 @@
+@RunWith(JUnit4.class)
+@SuppressWarnings("checkstyle:abbreviationaswordinname")
+public class ProductSearchIT {
+  private static final String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
+  private static final String COMPUTE_REGION = "us-west1";
+  private static final String GCS_URI =
+      "gs://java-docs-samples-testing/product-search/indexed_product_sets.csv";
+  private static final String PRODUCT_SET_ID = "indexed_product_set_id_for_testing";
+  private static final String PRODUCT_CATEGORY = "apparel";
+  private static final String PRODUCT_ID_1 = "indexed_product_id_for_testing_1";
+  private static final String PRODUCT_ID_2 = "indexed_product_id_for_testing_2";
+  private static final String IMAGE_URI_1 =
+      "gs://java-docs-samples-testing/product-search/shoes_1.jpg";
+  private static final String FILE_PATH_1 = "./resources/shoes_1.jpg";
+  private static final String FILTER = "style=womens";
+  private static final String BUCKET = "java-docs-samples-testing";
+  private ByteArrayOutputStream bout;
+  private PrintStream out;
+
+  @Before
+  public void setUp() throws Exception {
+
+    bout = new ByteArrayOutputStream();
+    out = new PrintStream(bout);
+    System.setOut(out);
+    ImportProductSets.importProductSets(PROJECT_ID, COMPUTE_REGION, GCS_URI);
+    bout.reset();
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    ProductManagement.deleteProduct(PROJECT_ID,COMPUTE_REGION,PRODUCT_ID_1);
+    ProductManagement.deleteProduct(PROJECT_ID,COMPUTE_REGION,PRODUCT_ID_2);
+    ProductSetManagement.deleteProductSet(PROJECT_ID, COMPUTE_REGION, PRODUCT_SET_ID);
+    System.setOut(null);
+  }
+
+  @Test
+  public void testGetSimilarProductsFile() throws Exception {
+    // Act
+    ProductSearch.getSimilarProductsFile(
+        PROJECT_ID, COMPUTE_REGION, PRODUCT_SET_ID, PRODUCT_CATEGORY, FILE_PATH_1, "");
+
+    // Assert
+    String got = bout.toString();
+    assertThat(got).contains(PRODUCT_ID_1);
+    assertThat(got).contains(PRODUCT_ID_2);
+  }
+
+  @Test
+  public void testGetSimilarProductsGcs() throws Exception {
+    // Act
+    ProductSearch.getSimilarProductsGcs(
+        PROJECT_ID, COMPUTE_REGION, PRODUCT_SET_ID, PRODUCT_CATEGORY, IMAGE_URI_1, "");
+
+    // Assert
+    String got = bout.toString();
+    assertThat(got).contains(PRODUCT_ID_1);
+    assertThat(got).contains(PRODUCT_ID_2);
+  }
+
+  @Test
+  public void testGetSimilarProductsFileWithFilter() throws Exception {
+    // Act
+    ProductSearch.getSimilarProductsFile(
+        PROJECT_ID, COMPUTE_REGION, PRODUCT_SET_ID, PRODUCT_CATEGORY, FILE_PATH_1, FILTER);
+
+    // Assert
+    String got = bout.toString();
+    assertThat(got).contains(PRODUCT_ID_1);
+    assertThat(got).doesNotContain(PRODUCT_ID_2);
+  }
+
+  @Test
+  public void testGetSimilarProductsGcsWithFilter() throws Exception {
+    // Act
+    //ImportProductSets.importProductSets(PROJECT_ID, COMPUTE_REGION, GCS_URI);
+    //bout.reset();
```

Comment:
Are these comments supposed to be commented or just deleted?

Reply:
Should be deleted. Fixing that now.